### PR TITLE
Add ability to version caches

### PR DESCRIPTION
**Pull request by larowlan**

### CODE REVIEW

File: src/jobs/build_backend.yml [hunk 1]
```diff
@@ -31,18 +31,22 @@ parameters:
...
 
 executor: <<parameters.executor>>
 
 steps:
   - checkout
   - restore_cache:
       keys:
-        - deps-composer-{{ arch }}-{{ checksum "composer.lock" }}
-        - deps-composer-{{ arch }}-
+        - deps-composer-{{ arch }}-{{ checksum "composer.lock" }}<<parameters.cache-version>>
```
Comment (kimpepper) [inline src/jobs/build_backend.yml:45]:
```suggestion
        - deps-composer-<<parameters.cache-version>>-{{ arch }}-{{ checksum "composer.lock" }}
```

File: src/jobs/build_backend.yml [hunk 2]
```diff
@@ -31,18 +31,22 @@ parameters:
...
+        - deps-composer-{{ arch }}-<<parameters.cache-version>>
```
Comment (kimpepper) [inline src/jobs/build_backend.yml:46]:
```suggestion
        - deps-composer-<<parameters.cache-version>>-{{ arch }}
```

File: src/jobs/build_backend.yml [hunk 3]
```diff
@@ -31,18 +31,22 @@ parameters:
...
   - steps: <<parameters.init>>
   - save_cache:
-      key: deps-composer-{{ arch }}-{{ checksum "composer.lock" }}
+      key: deps-composer-{{ arch }}-{{ checksum "composer.lock" }}<<parameters.cache-version>>
```
Comment (kimpepper) [inline src/jobs/build_backend.yml:49]:
```suggestion
      key: deps-composer-<<parameters.cache-version>>-{{ arch }}-{{ checksum "composer.lock" }}
```

File: src/jobs/patchy.yml [hunk 4]
```diff
@@ -22,8 +26,8 @@ steps:
   - checkout
   - restore_cache:
       keys:
-        - deps-composer-{{ arch }}-{{ checksum "composer.lock" }}
-        - deps-composer-{{ arch }}-
+        - deps-composer-{{ arch }}-{{ checksum "composer.lock" }}<<parameters.cache-version>>
```
Comment (kimpepper) [inline src/jobs/patchy.yml:29]:
```suggestion
        - deps-composer-<<parameters.cache-version>>-{{ arch }}-{{ checksum "composer.lock" }}
```

File: src/jobs/patchy.yml [hunk 5]
```diff
@@ -22,8 +26,8 @@ steps:
...
+        - deps-composer-{{ arch }}-<<parameters.cache-version>>
```
Comment (kimpepper) [inline src/jobs/patchy.yml:30]:
```suggestion
        - deps-composer-<<parameters.cache-version>>-{{ arch }}
```

File: src/jobs/patchy.yml [hunk 6]
```diff
@@ -12,6 +12,10 @@ parameters:
   ssh_fingerprint:
     description: The SSH fingerprint used for writing back to the project repo.
     type: string
+  cache-version:
+    description: An optional version number for cache dependencies, e.g. -v1
+    type: string
+    default: ~
```
Comment (kimpepper) [inline src/jobs/patchy.yml:18]:
```suggestion
    default: v1
```

**Review by kimpepper — CHANGES_REQUESTED** [review]
Just a few minor tweaks to the order of things.

**Comment by kimpepper** [comment]
@larowlan Should we tackle the npm deps at the same time?

**Comment by larowlan** [comment]
sure

File: src/jobs/build_backend.yml [hunk 7]
```diff
@@ -31,18 +31,22 @@ parameters:
     type: steps
     description: Run the the backend-end tests.
     default: []
+  cache-version:
+    description: An optional version number for cache dependencies, e.g. -v1
```
Comment (kimpepper) [inline src/jobs/build_backend.yml:35]:
```suggestion
    description: An optional version number for cache dependencies, e.g. v1
```

File: src/jobs/build_backend.yml [hunk 8]
```diff
@@ -31,18 +31,22 @@ parameters:
...
+    type: string
+    default: ~
```
Comment (kimpepper) [inline src/jobs/build_backend.yml]:
```suggestion
    default: v1
```

**Review by kimpepper — CHANGES_REQUESTED** [review]

File: src/jobs/build_frontend.yml [hunk 9]
```diff
@@ -42,18 +42,22 @@ parameters:
     type: steps
     description: Run the front-end unit tests.
     default: []
+  cache-version:
+    description: An optional version number for cache dependencies, e.g. -v1
```
Comment (kimpepper) [inline src/jobs/build_frontend.yml:46]:
```suggestion
    description: An optional version number for cache dependencies, e.g. v2
```

**Review by kimpepper — APPROVED** [review]

**Pull request merged**